Options.py

```diff
@@ -368,6 +401,48 @@ def __ne__(self, other):
     __hash__ = Option.__hash__  # see https://docs.python.org/3/reference/datamodel.html#object.__hash__
 
 
+class TextChoice(FreeText, Choice):
+    """Allows custom string input and offers choices. Choices will resolve to int and text will resolve to string"""
+
+    def __init__(self, value: typing.Union[str, int]):
+        if isinstance(value, str):
+            self.value: str = value
+        elif isinstance(value, int):
+            self.value: int = value
+        else:
+            raise TypeError(f"{value} is not a valid option for {self.__class__.__name__}")
```

Comment:
My personal opinion on this:
You give static typing, and for all allowed types the code run is the same. This code is never directly run by the user, only through `from_*`, which does str(user_input). So i see no reason to write it like this and it could be simplified to:
```python

    def __init__(self, value: typing.Union[str, int]):
        assert isinstance(value, str) or isinstance(value, int), f"{value} is not a valid option for {self.__class__.__name__}"
        self.value = value
```
which however will not do any type checking in the release build (run with `-O`).
(Assert statements should be used in a way that they never fail unless there is a bug in the code, because they get disabled in release. TypeError on the other hand should be thrown when type checking is required.)

Maybe @Berserker66 can clarify if we want to do those kinds of optimizations or not.